library/psa_crypto.c

```diff
@@ -1342,6 +1365,229 @@ psa_status_t psa_export_key( mbedtls_svc_key_id_t key,
     return( ( status == PSA_SUCCESS ) ? unlock_status : status );
 }
 
+#if defined(MBEDTLS_PSA_BUILTIN_KEY_TYPE_FFDH_KEY_PAIR) || \
+    defined(MBEDTLS_PSA_BUILTIN_KEY_TYPE_FFDH_PUBLIC_KEY)
+static psa_status_t psa_ffdh_set_prime_generator( size_t key_size,
+                                                  mbedtls_mpi *P,
+                                                  mbedtls_mpi *G )
+{
+    const unsigned char *dhm_P = NULL;
+    const unsigned char *dhm_G = NULL;
+    size_t dhm_size_P = 0;
+    size_t dhm_size_G = 0;
+    int ret = MBEDTLS_ERR_ERROR_CORRUPTION_DETECTED;
+
+    if( P == NULL && G == NULL )
+        return( PSA_ERROR_INVALID_ARGUMENT );
+
+    static const unsigned char dhm_P_2048[] =
+        MBEDTLS_DHM_RFC7919_FFDHE2048_P_BIN;
+    static const unsigned char dhm_P_3072[] =
+        MBEDTLS_DHM_RFC7919_FFDHE3072_P_BIN;
+    static const unsigned char dhm_P_4096[] =
+        MBEDTLS_DHM_RFC7919_FFDHE4096_P_BIN;
+    static const unsigned char dhm_P_6144[] =
+        MBEDTLS_DHM_RFC7919_FFDHE6144_P_BIN;
+    static const unsigned char dhm_P_8192[] =
+        MBEDTLS_DHM_RFC7919_FFDHE8192_P_BIN;
+    static const unsigned char dhm_G_2048[] =
+        MBEDTLS_DHM_RFC7919_FFDHE2048_G_BIN;
+    static const unsigned char dhm_G_3072[] =
+        MBEDTLS_DHM_RFC7919_FFDHE3072_G_BIN;
+    static const unsigned char dhm_G_4096[] =
+        MBEDTLS_DHM_RFC7919_FFDHE4096_G_BIN;
+    static const unsigned char dhm_G_6144[] =
+        MBEDTLS_DHM_RFC7919_FFDHE6144_G_BIN;
+    static const unsigned char dhm_G_8192[] =
+        MBEDTLS_DHM_RFC7919_FFDHE8192_G_BIN;
+
+    if( key_size <= 256 )
+    {
+        dhm_P = dhm_P_2048;
+        dhm_G = dhm_G_2048;
+        dhm_size_P = sizeof( dhm_P_2048 );
+        dhm_size_G = sizeof( dhm_G_2048 );
+    }
+    else if( key_size <= 384 )
+    {
+        dhm_P = dhm_P_3072;
+        dhm_G = dhm_G_3072;
+        dhm_size_P = sizeof( dhm_P_3072 );
+        dhm_size_G = sizeof( dhm_G_3072 );
+    }
+    else if( key_size <= 512 )
+    {
+        dhm_P = dhm_P_4096;
+        dhm_G = dhm_G_4096;
+        dhm_size_P = sizeof( dhm_P_4096 );
+        dhm_size_G = sizeof( dhm_G_4096 );
+    }
+    else if( key_size <= 768 )
+    {
+        dhm_P = dhm_P_6144;
+        dhm_G = dhm_G_6144;
+        dhm_size_P = sizeof( dhm_P_6144 );
+        dhm_size_G = sizeof( dhm_G_6144 );
+    }
+    else if( key_size <= 768 )
```

Comment:
duplication?

Reply:
Yes, duplicated condition. Needs to be fixed.
It's strange that compiler didn't point this.

Reply:
Fixed